src/hooks/useScrollTo.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect, useRef } from 'react';
+
+const DEFAULT_SCROLL_MARGIN = 300;
+
+const useScrollTo = (
```

Comment:
we're using `scrollTo`, because`.scrollIntoView` is not reliable enough for this use case.